app/javascript/components/subject/MoveToHousehold.js

```diff
@@ -0,0 +1,137 @@
+import { PropTypes } from 'prop-types';
+import axios from 'axios';
+import reportError from '../util/ReportError';
+
+class MoveToHousehold extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      updateDisabled: true,
+      showModal: false,
+      loading: false,
+      groupMembers: [],
+    };
+    this.toggleModal = this.toggleModal.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.submit = this.submit.bind(this);
+    this.getResponders = this.getResponders.bind(this);
+  }
+
+  toggleModal() {
+    let current = this.state.showModal;
+    this.setState({
+      updateDisabled: true,
+      showModal: !current,
+      loading: !current,
+    });
+    if (!current) {
+      this.getResponders();
+    }
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.id]: event.target.value, updateDisabled: false });
+  }
+
+  getResponders() {
+    axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+    axios({
+      method: 'get',
+      url: window.BASE_PATH + '/public_health/patients/self_reporting',
+      params: {},
+    })
+      .then(response => {
+        this.setState({
+          loading: false,
+          groupMembers: JSON.parse(response['data']['self_reporting']),
+        });
+      })
+      .catch(err => {
+        reportError(err);
+      });
+  }
+
+  submit() {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(window.BASE_PATH + '/patients/' + this.props.patient.id + '/update_hoh', {
+          new_hoh_id: this.state.hoh_selection,
+        })
+        .then(() => {
+          this.setState({ updateDisabled: false });
+          location.reload(true);
+        })
+        .catch(error => {
+          console.error(error);
+        });
+    });
+  }
+
+  createModal(title, toggle, submit) {
+    return (
+      <Modal size="lg" show centered>
+        <Modal.Header>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form>
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label className="nav-input-label">Select The New Monitoree That Will Respond For The Current Monitoree</Form.Label>
+                <Form.Label size="sm" className="nav-input-label">
+                  Note: The current monitoree will be moved into the selected monitorees household
+                </Form.Label>
+                <Form.Control as="select" className="form-control-lg" id="hoh_selection" onChange={this.handleChange} defaultValue={-1}>
+                  <option value={-1} disabled>
+                    --
+                  </option>
+                  {this.state?.groupMembers?.map((member, index) => {
+                    return (
+                      <option key={`option-${index}`} value={member.id}>
```

Comment:
Could we turn this into a datalist? Would be nice to be able to type/copy&paste if you know exactly who you are looking for instead of having to scroll.